src/test/java/io/lettuce/core/tracing/BraveTracingIntegrationTests.java

```diff
@@ -101,7 +101,9 @@ void pingWithTrace() {
         List<Span> spans = new ArrayList<>(BraveTracingIntegrationTests.spans);
 
         assertThat(spans.get(0).name()).isEqualTo("hello");
+        assertThat(spans.get(0).remoteEndpoint()).isNull();
```

Comment:
When I hade the code in `writeSingleCommand`, actually this was populated with `port` but after moving to `channelActive` it's empty. I guess since hello is made before a connection is, this could make sense though

Reply:
As fallback, it's fine to create a `Tracing.Endpoint` on demand in `writeSingleCommand` if `tracedEndpoint` is null. That should only happen for handshake commands and the risk of introducing too much performance drag isn't given here. I'll take care of this during the merge.